internal/proxy/proxy.go

```diff
@@ -89,34 +89,37 @@ type Proxy struct {
 	// Add callback functions at different stages
 	startCallbacks []func()
 	closeCallbacks []func()
+
+	logCtx context.Context
```

Comment:
It feels weird to storage a context with logger here. Maybe a logger will be enough?

Reply:
In this case yes, but if we want to pass a context to another function, we should know the log context. If I dont save the context, I will create this context every time I log infos.

Reply:
It wil be great if there's a better practice.

Reply:
`Proxy` already has a `ctx` field stored. (Not a good practice, BTW). Which context member shall we pass when invoke a utility function? My suggestion is: 
1. If proxy function needs a logger to pass on, with a context parameter, say `CreateCollection`, use context.WithValue to add the logger into context to pass on.
2. it other case, say `Init`, maybe we could either use ```context.WithValue(p.ctx, logger)```, or just use a Background one.

Reply:
`logCtx` is only used for saving a logger of a module. If a new grpc request arrives, we should get a ctx with logger by `WithValue`. Maybe grpc log context have other fields like requestID. I agree that storing a context is not a good practice, but if we store a logger in proxy struct and we should create a context with this logger every time we call other functions and this is a little dirty.